src/zarr/abc/store.py

```diff
@@ -297,9 +307,18 @@ async def set(self, value: Buffer, byte_range: ByteRangeRequest | None = None) -
 
     async def delete(self) -> None: ...
 
+    async def setdefault(self, default: Buffer) -> None: ...
+
 
 async def set_or_delete(byte_setter: ByteSetter, value: Buffer | None) -> None:
     if value is None:
         await byte_setter.delete()
     else:
         await byte_setter.set(value)
+
+
+async def setdefault(byte_setter: ByteSetter, value: Buffer | None) -> None:
```

Comment:
Is this used anywhere? Also, it seems to me like the behavior of this function doesn't match the intention of the `Store.setdefault` method

Reply:
Whoops, I ended up not implementing / using this. It's handled by `ByteSetter.setdefault`